services/static-webserver/client/source/class/osparc/data/model/Node.js

```diff
@@ -976,28 +976,37 @@ qx.Class.define("osparc.data.model.Node", {
       return true;
     },
 
-    requestStopNode: function() {
-      const preferencesSettings = osparc.Preferences.getInstance();
-      if (preferencesSettings.getConfirmStopNode()) {
-        const msg = this.tr("Do you really want Stop and Save the current state?");
-        const win = new osparc.ui.window.Confirmation(msg).set({
-          confirmText: this.tr("Stop")
-        });
-        win.center();
-        win.open();
-        win.addListener("close", () => {
-          if (win.getConfirmed()) {
-            const params = {
-              url: {
-                studyId: this.getStudy().getUuid(),
-                nodeId: this.getNodeId()
-              }
-            };
-            osparc.data.Resources.fetch("studies", "stopNode", params)
-              .then(() => this.stopDynamicService())
-              .catch(err => console.error(err));
+    requestStopNode: function(withConfirmationDialog=false) {
+      const self = this;
+      const stopService = function() {
```

Comment:
You can declare it as an arrow function `const stopService = () => {` to avoid having to save `this`. Arrow functions don't provide a new `this` binding.